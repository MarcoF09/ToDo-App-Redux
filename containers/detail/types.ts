import {
  NavigationParams,
  NavigationRoute,
  NavigationScreenProp
} from 'react-navigation'
import { Todo } from '../../types/globalTypes'

export interface Props {
  todo: Todo[]
  markAsNotDone: (todos: Todo[], index: number) => void
  markAsDone: (todos: Todo[], index: number) => void
  navigation: NavigationScreenProp<
    NavigationRoute<NavigationParams>,
    NavigationParams
  >
}

export type LayoutProps = Props
