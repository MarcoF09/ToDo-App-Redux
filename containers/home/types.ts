import { Animated } from 'react-native';
import { NavigationRoute, NavigationScreenProp } from 'react-navigation';
import { State, Todo } from '../../types/globalTypes';

export interface Props {
  todo: Todo[];
  navigation: NavigationScreenProp<NavigationRoute>;
  getToDoData: () => void;
  clearAllDone: (todos: Todo[]) => void;
  changeCheckBoxState: (todos: Todo[], index: number) => void;
  deleteItem: (todos: Todo[], index: number) => void;
  animateDeleteButton: (
    deleteButtonState: boolean,
    moveAnimation: Animated.ValueXY
  ) => void;
}

export interface HomeState extends State {
  deleteButtonState: boolean;
  moveAnimation: Animated.ValueXY;
}

export type LayoutProps = Props & HomeState;
