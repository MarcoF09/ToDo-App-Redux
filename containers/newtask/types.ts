import { NavigationRoute, NavigationScreenProp } from 'react-navigation';

interface Props {
  navigation: NavigationScreenProp<NavigationRoute>;
  addToDo: (title: string, description: string) => void;
  onChangeTitle: (text: string) => void;
  onChangeDescription: (text: string) => void;
  onFocusTitle: () => void;
  onBlurTitle: () => void;
  onFocusDescription: () => void;
  onBlurDescription: () => void;
}

interface NewTaskState {
  firstInputColor: string;
  secondInputColor: string;
  title: string;
  description: string;
}

export type LayoutProps = Props & NewTaskState;
