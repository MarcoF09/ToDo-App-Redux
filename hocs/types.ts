import React from 'react';

export type Hoc<TOuter = {}, TInner = TOuter> = (
  Component: React.ComponentType<TInner>
) => React.ComponentType<TOuter>;
export type ExtractTInner<T> = T extends Hoc<any, infer TInner>
  ? TInner
  : never;
