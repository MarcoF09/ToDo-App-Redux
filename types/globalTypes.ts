export interface Todo {
  id?: string;
  title: string;
  order?: number;
  description?: string;
  completed: boolean;
  url: string;
}

export interface ResponseBackend {
  title: string;
  order: number;
  completed: boolean;
  url: string;
}

export interface State {
  todo: Todo[];
}
