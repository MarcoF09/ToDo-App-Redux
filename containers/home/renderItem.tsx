import React from 'react'
import {
  Animated,
  ListRenderItemInfo,
  Text,
  TouchableWithoutFeedback,
  View
} from 'react-native'
import {
  AnimatedValue,
  NavigationRoute,
  NavigationScreenProp
} from 'react-navigation'
import { Item } from '../../components/item'
import { Todo } from '../../types/globalTypes'
import { styles } from './styles'

export const renderItem = (
  item: ListRenderItemInfo<Todo>,
  navigation: NavigationScreenProp<NavigationRoute>,
  todo: Todo[],
  moveAnimation: AnimatedValue,
  deleteItem: (todos: Todo[], index: number) => void,
  changeCheckBoxState: (todos: Todo[], index: number) => void
) => {
  return (
    <View testID={`item${item.index}`}>
      <Animated.View
        style={[styles.deleteContainer, moveAnimation.getLayout()]}
      >
        <TouchableWithoutFeedback
          testID="deleteButton"
          style={styles.buttonDelete}
          onPress={() => {
            deleteItem(todo, item.index)
          }}
        >
          <Text style={styles.buttonText}>-</Text>
        </TouchableWithoutFeedback>
      </Animated.View>

      <Item
        onPress={() => {
          navigation.push('Detail', {
            index: item.index
          })
        }}
        {...item.item}
        checkboxClick={() => {
          changeCheckBoxState(todo, item.index)
        }}
      />
    </View>
  )
}
