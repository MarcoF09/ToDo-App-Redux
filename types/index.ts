import { Action } from 'redux';
import { ActionsTypes } from '../actions/actionTypes';
import { Todo } from './globalTypes';

export interface ClearAllDoneSucessAction extends Action {
  type: ActionsTypes.CLEAR_ALL_DONE_SUCCESS;
  payload: {
    todo: Todo[];
  };
}

export interface ClearAllDoneFailAction extends Action {
  type: ActionsTypes.CLEAR_ALL_DONE_FAIL;
}

export interface ChangeCheckBoxStateActionSuccessAction extends Action {
  type: ActionsTypes.CHANGE_CHECKBOX_STATE_SUCCESS;
  payload: { todo: Todo[] };
}

export interface ChangeCheckBoxStateActionFailAction extends Action {
  type: ActionsTypes.CHANGE_CHECKBOX_STATE_FAIL;
}

export interface AddToDoSuccessAction extends Action {
  type: ActionsTypes.HANDLE_ADD_DATA_SUCCESS;
  payload: { title: string; description: string };
}

export interface AddToDoFailAction extends Action {
  type: ActionsTypes.HANDLE_ADD_DATA_FAIL;
}

export interface MarkAsNotDoneSuccessAction extends Action {
  type: ActionsTypes.HANDLE_MARK_AS_NOT_DONE_SUCCESS;
  payload: {
    todo: Todo[];
  };
}

export interface MarkAsNotDoneFailAction extends Action {
  type: ActionsTypes.HANDLE_MARK_AS_NOT_DONE_FAIL;
}

export interface MarkAsDoneSuccessAction extends Action {
  type: ActionsTypes.HANDLE_MARK_AS_DONE_SUCCESS;
  payload: {
    todo: Todo[];
  };
}

export interface MarkAsDoneFailAction extends Action {
  type: ActionsTypes.HANDLE_MARK_AS_DONE_FAIL;
}

export interface GetToDoDataSuccess extends Action {
  type: ActionsTypes.GET_TO_DO_DATA_SUCCESS;
  payload: {
    todo: Todo[];
  };
}

export interface GetToDoDataFail extends Action {
  type: ActionsTypes.GET_TO_DO_DATA_FAIL;
}

export interface DeleteItemSuccess extends Action {
  type: ActionsTypes.DELETE_ITEM_SUCCESS;
  payload: {
    todo: Todo[];
  };
}

export interface DeleteItemFail extends Action {
  type: ActionsTypes.DELETE_ITEM_FAIL;
}

export type Actions =
  | ClearAllDoneSucessAction
  | ClearAllDoneFailAction
  | ChangeCheckBoxStateActionSuccessAction
  | ChangeCheckBoxStateActionFailAction
  | AddToDoSuccessAction
  | AddToDoFailAction
  | MarkAsNotDoneSuccessAction
  | MarkAsNotDoneFailAction
  | MarkAsDoneSuccessAction
  | MarkAsDoneFailAction
  | GetToDoDataSuccess
  | GetToDoDataFail
  | DeleteItemSuccess
  | DeleteItemFail;
