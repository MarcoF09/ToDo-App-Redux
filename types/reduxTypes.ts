import { State } from '../types/globalTypes';

export interface Action<T = any> {
  type: T;
}

export interface AnyAction extends Action {
  // Allows any extra properties to be defined in an action.
  [extraProps: string]: any;
}

export type Dispatch<A extends Action = AnyAction> = <T extends A>(
  action: T
) => T;

export type Reducer<S = State, A extends Action = AnyAction> = (
  state: S | undefined,
  action: A
) => S;

export type ReducersMapObject<S = any, A extends Action = Action> = {
  [K in keyof S]: Reducer<S[K], A>
};

export interface Store<S = any, A extends Action = AnyAction> {
  dispatch: Dispatch<A>;
  getState(): S;
  replaceReducer(nextReducer: Reducer<S, A>): void;
}

export type ActionCreator<A> = (...args: any[]) => A;
