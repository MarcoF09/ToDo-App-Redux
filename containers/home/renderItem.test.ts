import { Animated } from 'react-native'
import { Todo } from 'types/globalTypes'
import { renderItem } from './renderItem'

const initialTodo: Todo[] = [
  { title: 'Title1', description: 'Description1', completed: false, url: '' },
  { title: 'Title2', description: 'Description2', completed: true, url: '' }
]
const navigation = {
  navigate: jest.fn(),
  push: jest.fn()
}
describe('should render the footer', () => {
  const mockedFunction = jest.fn()
  const moveAnimation = new Animated.ValueXY({ x: -100, y: 30 })

  it('renders correctly the item', () => {
    expect(
      renderItem(
        {
          item: initialTodo[0],
          index: 0,
          separators: {
            highlight: jest.fn(),
            unhighlight: jest.fn(),
            updateProps: jest.fn()
          }
        },
        navigation,
        initialTodo,
        moveAnimation,
        mockedFunction,
        mockedFunction
      )
    ).toMatchSnapshot()
  })
})
