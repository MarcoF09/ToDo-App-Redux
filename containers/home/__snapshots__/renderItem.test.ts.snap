// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the footer renders correctly the item 1`] = `
<View>
  <AnimatedComponent
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#c1131f",
          "borderRadius": 100,
          "display": "flex",
          "height": 20,
          "justifyContent": "center",
          "position": "absolute",
          "width": 20,
          "zIndex": 2,
        },
        Object {
          "left": -100,
          "top": 30,
        },
      ]
    }
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
      style={
        Object {
          "paddingTop": 24,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
          }
        }
      >
        -
      </Text>
    </TouchableWithoutFeedback>
  </AnimatedComponent>
  <Item
    checkboxClick={[Function]}
    completed={false}
    description="Description1"
    onPress={[Function]}
    title="Title1"
    url=""
  />
</View>
`;
